sdk/src/types/block/mana/allotment.rs

```diff
@@ -54,53 +64,131 @@ impl WorkScore for ManaAllotment {
     }
 }
 
-fn verify_mana<const VERIFY: bool>(mana: &u64, params: &ProtocolParameters) -> Result<(), Error> {
-    if VERIFY && *mana > params.mana_parameters().max_mana() {
+fn verify_mana<const VERIFY: bool>(mana: &u64) -> Result<(), Error> {
+    if VERIFY && *mana == 0 {
         return Err(Error::InvalidManaValue(*mana));
     }
 
     Ok(())
 }
 
-#[cfg(feature = "serde")]
-pub(super) mod dto {
-    use serde::{Deserialize, Serialize};
+pub(crate) type ManaAllotmentCount =
+    BoundedU16<{ *ManaAllotments::COUNT_RANGE.start() }, { *ManaAllotments::COUNT_RANGE.end() }>;
+
+/// A list of [`ManaAllotment`]s with unique [`AccountId`]s.
+#[derive(Clone, Debug, Eq, PartialEq, Deref, Packable)]
+#[packable(unpack_visitor = ProtocolParameters)]
+#[packable(unpack_error = Error, with = |e| e.unwrap_item_err_or_else(|p| Error::InvalidManaAllotmentCount(p.into())))]
+pub struct ManaAllotments(
+    #[packable(verify_with = verify_mana_allotments)] BoxedSlicePrefix<ManaAllotment, ManaAllotmentCount>,
+);
+
+impl ManaAllotments {
+    /// The minimum number of mana allotments of a transaction.
+    pub const COUNT_MIN: u16 = 0;
+    /// The maximum number of mana allotments of a transaction.
+    pub const COUNT_MAX: u16 = 128;
+    /// The range of valid numbers of mana allotments of a transaction.
+    pub const COUNT_RANGE: RangeInclusive<u16> = Self::COUNT_MIN..=Self::COUNT_MAX; // [1..128]
```

Comment:
```suggestion
    pub const COUNT_RANGE: RangeInclusive<u16> = Self::COUNT_MIN..=Self::COUNT_MAX;
```

Reply:
I think it's nice tbh, we have it for the other ranges as well

Reply:
all good